0019-glossary/0019-glossary.md

```diff
@@ -0,0 +1,248 @@
+When initiating an Interledger payment, the sender attaches an ILP packet to the local transfer to the connector. The packet is relayed by connectors and attached to each transfer that comprises the payment. In some cases, ledger protocols may define alternative ways to communicate the packet.
+Note that anyone can claim to have a certain Interledger address, there is no registry of them.
+Whether or not a payment ends up at the intended receiver is ultimately safe-guarded by the hashlock condition,
+not by enforcement of address ownership.
+See [IL-RFC 15](../0015-ilp-addresses/0015-ilp-addresses.md)
+
+## Interledger Architecture
+The [Interledger protocol stack](interledger-protocol-stack), and the design principles behind it.
+
+## Interledger Layer
+The lowest layer of the Interledger protocol stack (not counting the 'ledger layer' below it), consisting of
+the [ILP](#interledger-protocol-ilp) Interledger protocol, and the [ILQP](#ilqp) quoting protocol,
+
+## Interledger Module
+The part of a software application that processes ILP payments.
+Analoguous to the network card of an internet-connected computer.
+
+## Interledger Packet
+The binary data packet that is forwarded along all hops from sender to receiver, and which provides the receiver
+with metadata which the receiver may need to be willing and able to fulfill the payment.
+It also contains the destination account's address and destination amount, on which connectors will base their routing decisions -
+both in choosing the onward path, and in deciding whether their incoming amount was big enough, in comparison to the destination
+amount mentioned in the ILP packet, to even merit an [forward action](#forward-action) at all. See also [routing table](#routing-table)
+
+## Interledger Payment Request (IPR)
+A Transport Layer protocol in which the receiver generates the payment details and condition. The receiver does not share the secret used to generate the condition and fulfillment with the sender or anyone else, but the sender must ask the recipient to generate and share a condition before sending each payment. IPR is primarily useful for building non-repudiable application layer protocols, in which the sender's posession of the fulfillment proves to third parties that the sender has paid the receiver for a specific obligation.
+See [IL-RFC 11](../0011-interledger-payment-request/0011-interledger-payment-request.md).
+
+# Interledger Protocol (ILP)
+The core of the Interledger protocol suite, described in [IL-RFC 3](../0003-interledger-protocol/0003-interledger-protocol.md).
+Colloquially the whole Interledger stack is sometimes referred to as "ILP". Technically, however, the Interledger Protocol is only one layer in the
+[Interledger protocol stack](#interledger-protocol-stack).
+
+## Interledger protocol stack
+The stack consisting of the [Interledger layer](#interledger-layer),
+a choice of [transport layers](#transport-layer), and an [application layer](#application-layer).
```

Comment:
I think the stack refers to the whole set of protocols, not just when you choose one from each layer